MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/kits/basekits/Kit.java

```diff
@@ -35,95 +36,184 @@
 
-  public Kit(String name) {
-    setName(name);
-    setKey(name);
-  }
+    private HashMap<ItemStack, Integer> kitItems = new HashMap<>();
+    private ItemStack kitHelmet;
+    private ItemStack kitChestplate;
+    private ItemStack kitLeggings;
+    private ItemStack kitBoots;
 
-  public Kit(String name, String key) {
-    setName(name);
-    setKey(key);
-  }
+    protected Kit() {
+    }
+
+    public Kit(String name) {
+        setName(name);
+        setKey(name);
+    }
+
+    public Kit(String name, String key) {
+        setName(name);
+        setKey(key);
+    }
 
-  public abstract boolean isUnlockedByPlayer(Player p);
+    public abstract boolean isUnlockedByPlayer(Player p);
 
-  public boolean isUnlockedOnDefault() {
-    return unlockedOnDefault;
-  }
+    public boolean isUnlockedOnDefault() {
+        return unlockedOnDefault;
+    }
+
+    public void setUnlockedOnDefault(boolean unlockedOnDefault) {
+        this.unlockedOnDefault = unlockedOnDefault;
+    }
+
+    public void addKitItem(ItemStack item, Integer slot) {
+        kitItems.put(item, slot);
+    }
 
-  public void setUnlockedOnDefault(boolean unlockedOnDefault) {
-    this.unlockedOnDefault = unlockedOnDefault;
-  }
+    public HashMap<ItemStack, Integer> getKitItems() {
+        return kitItems;
+    }
 
-  /**
-   * @return main plugin
-   */
-  public PluginMain getPlugin() {
-    return plugin;
-  }
+    public void setKitItems(HashMap<ItemStack, Integer> kitItems) {
+        this.kitItems = kitItems;
+    }
 
-  /**
-   * @return config file of kits
-   */
-  public FileConfiguration getKitsConfig() {
-    return kitsConfig;
-  }
+    /**
+     * @return main plugin
+     */
+    public PluginMain getPlugin() {
+        return plugin;
+    }
 
-  public String getName() {
-    return name;
-  }
+    /**
+     * @return config file of kits
+     */
+    public FileConfiguration getKitsConfig() {
+        return kitsConfig;
+    }
 
-  public void setName(String name) {
-    if(name != null) {
-      this.name = name;
+    public void saveKitsConfig() {
+        ConfigUtils.saveConfig(plugin, kitsConfig, "kits");
+        kitsConfig = ConfigUtils.getConfig(plugin, "kits");
     }
-  }
 
-  public void setKey(String key) {
-    this.key = key;
-  }
+    public String getName() {
+        return name;
+    }
 
+    public void setName(String name) {
+        if (name != null) {
+            this.name = name;
+        }
+    }
 
-  public String getKey() {
-    if (key.equalsIgnoreCase("")) {
-      return name;
+    public void setKey(String key) {
+        this.key = key;
     }
-    return key;
-  }
 
-  public String[] getDescription() {
-    return description.clone();
-  }
+    public abstract void setupKitItems();
 
-  public void setDescription(String[] description) {
-    if(description != null) {
-      this.description = description.clone();
+    public String getKey() {
+        if (key.equalsIgnoreCase("")) {
```

Comment:
```suggestion
        if (key.isEmpty()) {
```

Reply:
Pretty  sure I didn't change that bit but hey 🤣 .

Reply:
yea, but why not if we're here